ptbcontrib/extended_keyboards/keyboards.py

```diff
@@ -0,0 +1,190 @@
+
+    @abstractmethod
+    def get_buttons(
+        self,
+    ) -> list[InlineKeyboardButton]:
+        """Just get flat list of buttons of the keyboard"""
+
+    @abstractmethod
+    def split(
+        self,
+        buttons_in_row: int,
+        empty_rows_allowed: bool = True,
+        strict: bool = False,
+    ) -> list[list[InlineKeyboardButton]]:
+        """
+        Split keyboard by N buttons in row.
+        Last row will contain remainder,
+        i.e. num of buttons in the last row maybe less than `buttons_in_row` parameter.
+
+        Possible enhancement:
+            keep_empty_rows: bool - keep empty rows in final keyboard if not enough buttons.
+            # Please create feature issue if you need it.
+        """
+
+
+class ExtendedInlineKeyboardMarkup(
+    InlineKeyboardMarkup,
+    IExtendedInlineKeyboardMarkup,
+):
+    """Popular keyboard actions"""
+
+    class Exceptions:
+        """Just a data container"""
+
+        class NotEnoughButtons(
+            Exception,
+        ):
+            """Exception raised when there are not enough buttons to fill the keyboard."""
+
+            def __init__(
+                self,
+                inline_keyboard: tuple[tuple[InlineKeyboardButton, ...], ...],
+                buttons: list[InlineKeyboardButton,],
+                buttons_in_row: int,
+            ) -> None:
+                super().__init__(
+                    "Total count of buttons not enough to fill the keyboard by equal rows "
+                    f"({len(buttons)} collected and "
+                    f"{buttons_in_row * len(inline_keyboard, )} required."
+                )
+
+        class EmptyRowsDisallowed(
+            Exception,
+        ):
+            """Exception raised when empty rows are not allowed in the keyboard."""
+
+            def __init__(
+                self,
+            ) -> None:
+                super().__init__(
+                    "Result num of rows less that original num of rows in the keyboard."
+                )
+
+    def to_list(
+        self,
+    ) -> list[list[InlineKeyboardButton]]:
+        """Just convert to list. By default, inline_keyboard is tuple[tuple[...]]"""
+        list_keyboard = [list(row) for row in self.inline_keyboard]
+        return list_keyboard
+
+    def find_btn_by_cbk(
+        self,
+        cbk: str,
+        strict: bool = True,
+    ) -> tuple[InlineKeyboardButton, int, int] | None:
+        """Returns buttons, row index, column index if found, None otherwise"""
+        for row_index, row in enumerate(self.inline_keyboard):
+            for column_index, _ in enumerate(row):
+                if strict:
+                    condition = cbk == self.inline_keyboard[row_index][column_index].callback_data
+                else:  # "in" to keep ability to match by prefix and some cbk key if need it
+                    condition = cbk in self.inline_keyboard[row_index][column_index].callback_data
```

Comment:
this will throw an exception in case `callback_data` is not a string, which can be the case if arbitrary `callback_data` is used. Please consider handling that case or at least documenting this limitation.